grunttasks/xss-mitigate.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var xss = require('xss');
+
+module.exports = function (grunt) {
+  grunt.registerTask('xss-flag', 'Run the XSS parser on the generated l10n files', [
+    'l10n-json-to-html',
+    'xss-parse'
+    ]
+  );
+
+  grunt.registerMultiTask('xss-parse', 'Parse HTML files to check for XSS vectors', function () {
+    var templateRegex = /\%\(\w+\)s/;
+    this.files.forEach(function (file) {
+      // iterate through each HTML file in the .tmp folder
+      var src = file.src[0];
+      var contents = grunt.file.read(src);
+      // whitelist of allowed tags and attributes, nothing else is allowed
+      var errorFound = false;
+      var options = {
+        whiteList: {
+          a: ['href', 'id', 'style'],
+          em: [],
+          span: ['id', 'tabindex'],
+          strong: []
+        },
+        onTagAttr: function (tag, name, value, isWhiteAttr) {
+          // On encountering any tag, check if the tag has a valid attribute.
+          // If the attribute is valid, check for its value, flag if illegal
+          // Illegal includes any mention of the string 'javascript', case insensitive
+          // Templated strings are allowed i.e strings of the form '%(termsURI)s'
+          // If these are considered to be not secure, we should change
+            if (name !== 'style' && isWhiteAttr && xss.safeAttrValue(tag, name, value) === '' && ! templateRegex.test(value)) {
+              grunt.log.error('%s: INVALID VALUE FOR ATTRIBUTE <%s %s="%s">', src, tag, name, value);
+              return '';
+            }
+        },
+        onIgnoreTag: function (tag, html, options) {
+          // On encountering any illegal tag, flag it
+          grunt.log.error('%s: INVALID TAGS found <%s>', src, tag);
+          errorFound = true;
+          return '';
+        },
+        onIgnoreTagAttr: function (tag, name, value, isWhiteAttr) {
+          // On encountering any illegal attribute, flag it
+          grunt.log.error('%s: INVALID TAG ATTRIBUTES FOUND <%s %s="">', src, tag, name);
+          errorFound = true;
+          return '';
+        }
+      };
+      contents = xss(contents, options);
+      if (errorFound) {
+        // if an error was found in this file, write it back out.
+        grunt.file.write(src, contents);
+      }
+    });
+    if (this.errorCount !== 0) {
```

Comment:
@shane-tomlinson It's a Grunt task thing: http://gruntjs.com/api/inside-tasks#this.errorcount

> _"The number of `grunt.log.error()` calls that occurred during this task. This can be used to fail a task if errors were logged during the task."_